moveit_simple/test/moveit_simple.cpp

```diff
@@ -50,50 +50,185 @@ TEST(MoveitSimpleTest, reachability)
+
+  const std::unique_ptr<moveit_simple::TrajectoryPoint> traj_point_joint2 =
+     std::unique_ptr<moveit_simple::TrajectoryPoint>
+     (new moveit_simple::JointTrajectoryPoint(joint1, 3.0, "traj_joint_point2"));
+
+  const std::unique_ptr<moveit_simple::TrajectoryPoint> traj_point_cart1 =
+     std::unique_ptr<moveit_simple::TrajectoryPoint>
+     (new moveit_simple::CartTrajectoryPoint(pose1, 4.0, "traj_point_cart1"));
+
+  const std::unique_ptr<moveit_simple::TrajectoryPoint> traj_point_cart2 =
+     std::unique_ptr<moveit_simple::TrajectoryPoint>
+     (new moveit_simple::CartTrajectoryPoint(pose3, 5.0, "traj_point_cart2"));
+
+  EXPECT_NO_THROW(robot2.addTrajPoint(TRAJECTORY_NAME, "home",      0.5));
+  // Populating points for further use and a fixed known point to start from
+  EXPECT_TRUE(robot2.toJointTrajectory(TRAJECTORY_NAME,points));
+  EXPECT_EQ(points.size(),2);
+  // joint interpolation towards joint point
+  EXPECT_TRUE(robot2.jointInterpolation(traj_point_joint1, points, (unsigned int) 10));
+  EXPECT_EQ(points.size(),13);
+  // Cartesian Interpolation towards cartesian point
+  EXPECT_TRUE(robot2.cartesianInterpolation(traj_point_cart1, points, (unsigned int) 0));
+  EXPECT_EQ(points.size(),14);
+  // Cartesian Interpolation between pose1 and pose3 is not possible but joint interpolation is
+  EXPECT_FALSE(robot2.cartesianInterpolation(traj_point_cart2, points, (unsigned int) 5));
+  EXPECT_EQ(points.size(),14);
+  // Joint Interpolation towards cartesian point
+  EXPECT_TRUE(robot2.jointInterpolation(traj_point_cart2, points, (unsigned int) 5));
+  EXPECT_EQ(points.size(),20);
+  // Cartesian Interpolation towards joint point
+  EXPECT_TRUE(robot2.jointInterpolation(traj_point_joint2, points, (unsigned int) 15));
+  EXPECT_EQ(points.size(),36);
+}
+
+
 TEST(MoveitSimpleTest, kinematics)
 {
   moveit_simple::Robot robot(ros::NodeHandle(), "robot_description", "manipulator");
   const Eigen::Affine3d pose = Eigen::Affine3d::Identity();
   Eigen::Affine3d pose1;
-  std::vector<double> joint_point(6,0);
-  std::vector<double> seed;
+  Eigen::Affine3d pose2;
+  std::vector<double> joint_point1(6,0);
+  std::vector<double> joint_point2;
+  std::vector<double> joint_point3;
+  std::vector<double> seed(6,0.1);
   ros::Duration(2.0).sleep();  //wait for tf tree to populate
 
-  EXPECT_TRUE(robot.getPosition(joint_point, pose1));
-  EXPECT_TRUE(robot.getJointSolution(pose1, 3.0, seed, joint_point));
-  EXPECT_FALSE(robot.getJointSolution(pose, 3.0, seed, joint_point));
+  EXPECT_TRUE(robot.getPose(joint_point1, pose1));
+  EXPECT_TRUE(robot.getJointSolution(pose1, 3.0, seed, joint_point2));
+  EXPECT_FALSE(robot.getJointSolution(pose, 4.0, seed, joint_point3));
+  EXPECT_TRUE(robot.getPose(joint_point2, pose2));
+
+  double error_pose = (pose1.translation() - pose2.translation()).norm() +
+                      (pose1.linear() - pose2.linear()).norm();
+  EXPECT_NEAR(error_pose, 0.0, 1e-2);
```

Comment:
If this test fails, it will be nearly impossible to know why.  Can you print out the two eigen transforms (the eigen matrix support the iostream operator, so this shouldn't be too hard).  You can print it out to ROS_DEBUG if it creates too much noise.